enumerable_project.rb

```diff
@@ -0,0 +1,387 @@
+# rubocop:disable Metrics/CyclomaticComplexity
+
+# rubocop:disable Metrics/PerceivedComplexity
+
+# rubocop:disable Metrics/ModuleLength
+
+module Enumerable
+  def my_each
+    if self.class == Array
+
+      0.upto(self.length - 1) do |i|
+        yield(self[i])
+      end
+
+    elsif self.class == Hash
+
+      for key in self
+
+        yield(key[0].to_s, key[1])
+
+      end
+
+    end
+  end
+
+  def my_each_with_index
+    if self.class == Array
+
+      0.upto(self.length - 1) do |i|
+        yield(self[i], i)
+      end
+
+    elsif self.class == Hash
+
+      i = 0
+
+      for key in self
+
+        yield([key[0].to_s, key[1]], i)
+
+        i+=1
+
+      end
+
+    end
+  end
+
+  def my_select
+    if self.class == Array
+
+      arr = []
+
+      0.upto(self.length-1) do |i|
+        arr << self[i] if yield(self[i])
+      end
+
+      return arr
+
+    elsif self.class == Hash
+
+      hash = {}
+
+      for key in self
+
+        hash[key[0]] = key[1] if yield(key[0], key[1])
+
+      end
+
+      return hash
+
+    end
+  end
+
+  def my_all?
+    if self.class == Array
+
+      0.upto(self.length-1) do |i|
+        return false unless yield(self[i])
+      end
+
+      return true
+
+    elsif self.class == Hash
+
+      for key in self
+
+        return false unless yield(key[0], key[1])
+
+      end
+
+      return true
+
+    end
+  end
+
+  def my_none?
+    if self.class == Array
+
+      0.upto(self.length-1) do |i|
+        return false if yield(self[i])
+      end
+
+      return true
+
+    elsif self.class == Hash
+
+      for key in self
+
+        return false if yield(key[0], key[1])
+
+      end
+
+      return true
+
+    end
+  end
```

Comment:
- [ ] should have working `my_any?` identical and returns the same thing as ruby's [`any?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-any-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_any.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

 #### my_any?  
(You had my_any? in the previous version but you deleted it in this ones let's bring it back with all the corrections)

- [ ] when no block or argument is given returns true if at least one of the collection is not false or nil

```ruby     
[nil, false, true, []].my_any? == [nil, false, true, []].any? => true
```
- Yours returns an NoMethodError
![image](https://user-images.githubusercontent.com/52209067/78953131-0f156e80-7a95-11ea-857c-9615483b096d.png)

- Real any?
![image](https://user-images.githubusercontent.com/52209067/77689425-3f262300-6f67-11ea-88a9-2885ebab08d4.png)

- [ ] when a class is passed as an argument returns true if at least one of the collection is a member of such class

```ruby
[1, 2.5, 'a', 9].my_any?(Integer) == [1, 2.5, 'a', 9].any?(Integer) => true
```
- Yours is retrieving a NoMethodError
![image](https://user-images.githubusercontent.com/52209067/78953146-248a9880-7a95-11ea-939b-9606890d4285.png)

- Real any?
![image](https://user-images.githubusercontent.com/52209067/77689595-8f04ea00-6f67-11ea-9f51-4536c3f8049b.png)

- [ ] when a Regex is passed as an argument returns false if none of the collection matches the Regex

```ruby
%w[dog door rod blade].my_any?(/z/) == %w[dog door rod blade].any?(/z/) => false
```
- Yours retrieves a NoMethodError
![image](https://user-images.githubusercontent.com/52209067/78953172-3835ff00-7a95-11ea-8637-e1877f31c5e2.png)

- Real any?
![image](https://user-images.githubusercontent.com/52209067/77689742-cf646800-6f67-11ea-9e3e-eec7a960fe38.png)

- [ ] when a pattern other than Regex or a Class is given returns false if none of the collection matches the pattern

```ruby
[3,4,7,11].my_any?(3) should be equal to this [3,4,7,11].any?(3) => true
```
- Yours retrieves a NoMethodError
![image](https://user-images.githubusercontent.com/52209067/78953202-4f74ec80-7a95-11ea-81de-de9236646189.png)

- Real any?
![image](https://user-images.githubusercontent.com/52209067/77689883-10f51300-6f68-11ea-8a4c-319977bb3842.png)

- [ ] should have working `my_none?` identical and returns the same thing as ruby's [`none?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-none-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_none.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

#### my_none?

- [ ] when no block or argument is given returns true only if none of the collection members is true
```ruby
[nil, false, true, []].my_none? == [nil, false, true, []].none? => false
```
- Yours is always expecting a block
![image](https://user-images.githubusercontent.com/52209067/78953303-a5499480-7a95-11ea-8baf-d1d332e4ec99.png)

- Real none?
![image](https://user-images.githubusercontent.com/52209067/77690365-cd4ed900-6f68-11ea-9933-f73b0091208d.png)

- [ ] when a class is passed as an argument returns true if none of the collection is a member of such class
```ruby
[1, 2.5, 'a', 9].my_none?(Integer) == [1, 2.5, 'a', 9].none?(Integer) => false
```
- Yours is retrieving an ArgumentError
![image](https://user-images.githubusercontent.com/52209067/78953323-b98d9180-7a95-11ea-9583-d3c731160ff0.png)

- Real none?
![image](https://user-images.githubusercontent.com/52209067/77690539-130ba180-6f69-11ea-8bd0-6ca061ab05d2.png)

- [ ] when a Regex is passed as an argument returns true only if none of the collection matches the Regex
```ruby
%w[dog door rod blade].my_none?(/z/) == %w[dog door rod blade].none?(/z/) => true
```
- Yours is retrieving an ArgumentError
![image](https://user-images.githubusercontent.com/52209067/78953356-d0cc7f00-7a95-11ea-8f9f-7e797e74928d.png)

- Real none?
![image](https://user-images.githubusercontent.com/52209067/77690722-65e55900-6f69-11ea-8881-9afe27d263dc.png)

- [ ] when a pattern other than Regex or a Class is given returns true only if none of the collection matches the pattern
```ruby
[3,4,7,11].my_none?(3) == [3,4,7,11].none?(3) => false
```
- Yours is retrieving an ArgumentError
![image](https://user-images.githubusercontent.com/52209067/78953395-e8a40300-7a95-11ea-8cf7-0a2b3051c38f.png)

- Real none?
![image](https://user-images.githubusercontent.com/52209067/77690915-a9d85e00-6f69-11ea-81b3-7adb07b281ca.png)

[OPTIONAL]

I think we could get the block of my_none? from my_any? we should only do something like !my_any? for my_none? to work. For example:

  def my_none?(arg = nil, &block)
    !my_any?(arg, &block)
  end

To make this work we have to use block.call(element) instead of yield(element) in my_any?